source/facter/3.9/custom_facts.md

```diff
@@ -161,14 +161,17 @@ An example of the confine statement would be something like the following:
 Facter.add(:powerstates) do
   confine :kernel => 'Linux'
   setcode do
+    # Put your fact code here.
     Facter::Core::Execution.execute('cat /sys/power/states')
   end
 end
 ```
 
+> Note: As shown in the example above, put everything underneath the `confine` statement and inside the `setcode` block of your fact. Otherwise anything before it will be executed regardless of your confine options.
```

Comment:
I'll comment on this because I think this was a quoted by me on Slack.
It's more accurate to say that everything *inside* the `setcode` block is confined, and everything outside the `setcode` block is not confined.

The word "underneath" is vague and doesn't technically mean anything in this context.